homeassistant/components/roku/remote.py

```diff
@@ -50,8 +50,9 @@ async def async_turn_on(self, **kwargs) -> None:
     @roku_exception_handler
     async def async_turn_off(self, **kwargs) -> None:
         """Turn the device off."""
-        await self.coordinator.roku.remote("poweroff")
-        await self.coordinator.async_request_refresh()
+        if self.is_on:
```

Comment:
Generally we don't do this because what if the state is out of sync?

If a user doesn't want to call turn off, they can add a condition or use a scene instead.

Reply:
I generally agree, but it seems odd in this specific circumstance that an error is thrown, cluttering the logs. It took me quite a bit of debugging (as an end user) to figure out when/why this *seemingly* random timeout was occurring.  I'm not sure if the issue could be fixed upstream; I've never actually done any python development before. I have like 4 rokus in various automations and it felt very cumbersome to specifically check state in each of those to avoid these errors.  I also thought about just updating the log level for the roku component, but didn't want to miss out on other potentially *real* problems.

Perhaps an alternative solution would be to throw a more specific error that could be individually filtered out via a log regex?  I'm not sure how feasible that is, being completely inexperienced in Python, or if that is a viable solution.

Reply:
since this is a polling integration, it can be up to 10s stale. we could maybe suppress timeout error for power off since if it times out chances are the device is already off.